src/main/java/rx/internal/operators/OperatorSwitchIfEmpty.java

```diff
@@ -125,4 +89,36 @@ public void onNext(T t) {
             child.onNext(t);
         }
     }
+    
+    private static final class AlternateSubscriber<T> extends Subscriber<T> {
+        
+        private final ProducerArbiter arbiter;
+        private final Subscriber<? super T> child;
+
+        AlternateSubscriber(Subscriber<? super T> child, ProducerArbiter arbiter) {
+            this.child = child;
+            this.arbiter = arbiter;
+        }
+        
+        @Override
+        public void setProducer(final Producer producer) {
+            arbiter.setProducer(producer);
+        }
+
+        @Override
+        public void onCompleted() {
+            child.onCompleted();
+        }
+
+        @Override
+        public void onError(Throwable e) {
+            child.onError(e);
+        }
+
+        @Override
+        public void onNext(T t) {
+            child.onNext(t);
+            arbiter.produced(1);
```

Comment:
I think this ```produced(1)``` is unnecessary: sure the arbiter's value will grow indefinitely, but there won't be any further ```setProducer``` on it thus conserving the requested/delivered amount is less important.